### PR TITLE
GSW-569 feat: get `quotes` as string parameters

### PR DESCRIPTION
**Pull request by r3v4s**
- router path will be calculated from off-chain
- therefore, some of router contract parameters need change its type to primitive

### CODE REVIEW

**Review by notJoon — CHANGES_REQUESTED** [review]

**Comment by notJoon** [comment]
Everything else seems to be fine 👍

File: router/_RPC_call.gno [hunk 1]
```diff
@@ -8,187 +8,103 @@ import (
...
 
-	rr, err := json.Marshal(rpcReturnBestSwap)
+	rr, err := json.Marshal(estimatedRoutes)
 	if err != nil {
-		panic(ufmt.Sprintf("[ROUTER] _RPC_call.gno__BestSwapDry() || json.Marshal error with %v", err))
+		panic(ufmt.Sprintf("[ROUTER] _RPC_call.gno__EstimateRouteQuotes() || json.Marshal error with %v", err))
 	}
 
 	return string(rr)
 }
```
[the remark this answers is not in the document]
Comment (r3v4s) [reply router/_RPC_call.gno:110]: @notJoon like your suggestion with `handleMultiSwap` helper.

About the distinct type `SwapType`, even it has underlying string type, outer will be `SwapType` which isn't primitive. I'm not sure if user can call function(with gnokey or adena) that takes parameter other than primitive type.


I'll do some test and if it fails, will only merge `handleMultiswap`

[the remark this answers is not in the document]
Comment (r3v4s) [reply router/_RPC_call.gno:110]: @notJoon SwapType looking GOOD.
```
# test that the example packages directory is loaded and usable.

## start a new node
gnoland start

gnokey maketx addpkg -pkgdir $WORK -pkgpath gno.land/r/primi -gas-fee 1000000ugnot -gas-wanted 2000000 -broadcast -chainid=tendermint_test test1
stdout OK!

## execute OrigInc
gnokey maketx call -pkgpath gno.land/r/primi -func OrigInc -args 123 -gas-fee 1000000ugnot -gas-wanted 2000000 -broadcast -chainid=tendermint_test test1
stdout '(123 int)'

## execute MyInc
gnokey maketx call -pkgpath gno.land/r/primi -func MyInc -args 456 -gas-fee 1000000ugnot -gas-wanted 2000000 -broadcast -chainid=tendermint_test test1
stdout '(456 gno.land/r/primi.myInt)'

## execute OrigString
gnokey maketx call -pkgpath gno.land/r/primi -func OrigString -args "qwe" -gas-fee 1000000ugnot -gas-wanted 2000000 -broadcast -chainid=tendermint_test test1
stdout 'qwe'

## execute MyString
gnokey maketx call -pkgpath gno.land/r/primi -func MyString -args "zxc" -gas-fee 1000000ugnot -gas-wanted 2000000 -broadcast -chainid=tendermint_test test1
stdout 'zxc'

-- gno.mod --
module gno.land/r/primi

-- primi.gno --
package primi

type myInt int
type myString string

func OrigInc(x int) int {
	println("orig x is:", x)
	return x
}

func MyInc(x myInt) myInt {
	println("my x is:", x)
	return x
}

func OrigString(s string) string {
	println("orig string is:", s)
	return s
}

func MyString(s myString) myString {
	println("my string is:", s)
	return s
}
```

[the remark this answers is not in the document]
Comment (r3v4s) [reply router/_RPC_call.gno:110]: @notJoon unfortunately, we can pass `ExactIn` as typed parameter(not value parameter).
So swap type has to stay in old way ( handle*Swap is commited )

**Review by notJoon — APPROVED** [review]

**Pull request merged**